### PR TITLE
Add CAS and CAS library

### PR DESCRIPTION
**Pull request by hharchani**
Also, loaded the composer's autoload in root index.php

### CODE REVIEW

**Comment by polybuildr** [comment]
We're using the imperative mood for commits. Could you please update? :P

File: src/app/libraries/cas_lib.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    function __construct() {
+        if (! self::$initialized) {
+            global $cfg;
+            phpCAS::client(
+                CAS_VERSION_2_0,
+                $cfg['cas_host'],
+                $cfg['cas_port'],
+                $cfg['cas_context']
+            );
+            phpCAS::setNoCasServerValidation();
```
Comment (nisargjhaveri) [inline src/app/libraries/cas_lib.php]: We might also want to give a certificate here.

Comment (hharchani) [inline src/app/libraries/cas_lib.php]: might? toh jab denge tab add kar lenge, nahi?

**Comment by polybuildr** [comment]
When you squash this commit, please put in the "fixes T39" trigger somewhere in your commit body so that we can link this with phab.

**Comment by hharchani** [comment]
@nisargjhaveri Any thing left here? If not, you can merge it.

File: src/app/libraries/cas_lib.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                $cfg['cas_host'],
+                $cfg['cas_port'],
+                $cfg['cas_context']
+            );
+            // For production use set the CA certificate that is the issuer
+            // of the cert on the CAS server and uncomment the line below
+            //phpCAS::setCasServerCACert($cfg['cas_server_ca_cert']);
+
+            // For quick testing, SSL validation of the CAS server can be disabled.
+            phpCAS::setNoCasServerValidation();
```
Comment (nisargjhaveri) [inline src/app/libraries/cas_lib.php]: This should be set automatically only when cert file is not given.

**Comment by nisargjhaveri** [comment]
And, squash, please?

**Comment by polybuildr** [comment]
And in the squashed commit please use the phab triggers.

**Comment by hharchani** [comment]
What's squash?

**Comment by polybuildr** [comment]
http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git/

First two answers are very useful.

**Comment by hharchani** [comment]
{Fix T39}

**Comment by nisargjhaveri** [comment]
Looks fine.

**Pull request merged**